x-pack/plugins/viz_editor/public/common/lib/vis_model.ts

```diff
@@ -19,25 +21,9 @@ export interface IndexPattern {
   fieldFormatMap?: string;
 }
 
-export interface DateHistogramOperation {
-  op: 'date_histogram';
-  arg: {
-    field: string;
-    interval: string;
-  };
-}
-
-export interface SumOperation {
-  op: 'sum';
-  arg: string;
-}
-
-export type QueryColumn = DateHistogramOperation | SumOperation;
-
 export interface VisModelQuery {
-  indexPattern: string;
   select: {
-    [id: string]: QueryColumn;
+    [id: string]: SelectOperation;
```

Comment:
Are we supposed to order operations by their IDs, or should you change the data type to `select: SelectOperation[]`?

Reply:
I think we either need to use an array, or we need to add a sort / order field. The disadvantage of arrays is that if you reorder them / add / remove, etc, then everyone who's referring to them need to update their indices. Hence, in normalized Redux state, you rarely see arrays, but you see objects, and sort-order is kept either as a field, or as an array of ids. I honestly don't know what we should do, but I suspect we'll have a lot of edge cases if we use an array, so I vote keeping this as a dictionary.